api/lib/domain/events/handle-certification-scoring.js

```diff
@@ -64,38 +67,30 @@ async function _calculateCertificationScore({
 async function _saveResult({
   assessment,
   assessmentScore,
-  partnerCertifications,
+  domainTransaction,
   assessmentResultRepository,
   certificationCourseRepository,
-  certificationPartnerAcquisitionRepository,
   competenceMarkRepository,
 }) {
-  const assessmentResult = await _createAssessmentResult({ assessment, assessmentScore, assessmentResultRepository });
+  const assessmentResult = await _createAssessmentResult({
+    assessment,
+    assessmentScore,
+    assessmentResultRepository,
+    domainTransaction
+  });
 
   await Promise.map(assessmentScore.competenceMarks, (competenceMark) => {
```

Comment:
Ne pas lancer un nombre indéterminé de traitements en parallèle (ici pour chaque élément de `assessmentScore.competenceMarks` une connexion à la base de données sera utilisée ce qui une utilisation disproportionnée compte tenu du fait que le processus entier est limité à 20 connexions simultanées). Utiliser `bluebird.mapSeries` ou `for(of) { await… }`.

Reply:
@jonathanperret Et si on exécute toutes les méthodes de répo à l'aide d'une même transaction, là par contre on n'utilise qu'une seule et même connexion c'est ça? Devrait-on démocratiser cette usage ?

Reply:
Comme on ne peut pas lancer plusieurs requêtes en parallèle dans une transaction, si `knex` est obligé de mettre en attente celles qu'on aurait lancées en parallèle. En pratique ce qui sera envoyé à la base sera la même chose (une requête à la fois) qu'on fasse un `bluebird.map(…)` ou un `bluebird.mapSeries(…)`. Autant donc utiliser une version sérielle (`mapSeries` ou `for(of) { await… }` qui rend plus explicite le fait qu'on exécute les requêtes une à une, et a le bon comportement même hors transaction.

Par ailleurs, systématiser l'utilisation de transactions, je ne serais pas contre.